### ISSUE TITLE
Posibility to get the old Layout back?

### ISSUE THREAD

**Issue by Tommy-LSA** [acomminos/Plumble-Legacy#109]
Large Server  are not handable with the actual layout. I can only recommend again having a look on mumble.piratenpartei-nrw.de You will see that the actual layout is not really sufficient . The List moves back and force because each second users change the channel, login or logout.
If the actual design is final the users of this server have to fork an earlier state of the project. I got a lot of complains because I have recommend plumble to hundreds of people which come back to me now and ask why the hell it has changed.

**Comment by Zuko** [comment]
Plumble should allow to collapse/expand channels and update only numbers
how many users are i other channels

2013/4/12 Tommy-LSA notifications@github.com

**Comment by Tommy-LSA** [comment]
No, all hundreds of channels are expanded and if I click on a channel I join this channel. How collapse should work? Tested on CM 9.1.0 on SGS and Android 4.1.2 on Galaxy Note2

**Comment by Zuko** [comment]
You misunderstood me, Plumble should have that option but it doesn't.
12 kwi 2013 23:10, "Tommy-LSA" notifications@github.com napisał(a):

**Comment by Tommy-LSA** [comment]
ahh okay. As it is now plumble is useless for me and a lot of other people who had payed for :-/

**Comment by Zuko** [comment]
Compile older version and wait for a "fix"
12 kwi 2013 23:22, "Tommy-LSA" notifications@github.com napisał(a):

**Comment by Tommy-LSA** [comment]
I'm able to compile but because I stuck in other projects I have bought the Playstore Version to get the updates. Never thought that it get unhandlable one time :-/

**Comment by Zuko** [comment]
Bigger servers, bigger problems ;] even PC version isn't "properly"
prepared to big servers

2013/4/12 Tommy-LSA notifications@github.com

**Comment by acomminos** [comment]
Hi,

I'd much rather fix your issues with the new design than return to the old. I will check out your server this weekend and try my best to resolve the scrolling issues etc. Sorry for the inconvenience.

**Comment by Tommy-LSA** [comment]
As Zuko mentioned, if the list would be collapsed as default and would have a reduced item height it "could be" a solution. With the actual solution you are completely lost on large servers.
Actually I have removed the Version from store and use an old selfcompiled one. So would great inform us if you have news on this because we get no updates anymore.

**Comment by acomminos** [comment]
Will do. Please test out the latest version, I added a few conveniences that should make it a bit more liveable (autoscroll to user position, quick scroll, favourites will scroll you over to the channel). When I visited your server I didn't have too much trouble.

**Comment by acomminos** [comment]
Having collapsed channels is a bit difficult to do with the ExpandableListView we use for the channel display. It doesn't truly support multiple levels of nesting.

**Comment by Tommy-LSA** [comment]
What do you think about a third screen with users between channels and chat?

**Comment by acomminos** [comment]
I suppose that would work as a last resort. What do you think of incorporating https://github.com/kedzie/tree-view-list-android to support multi-level nesting?

**Comment by Tommy-LSA** [comment]
treeviewlist sounds good. Ihave used such control often under .NET. Give it a try

**Comment by acomminos** [comment]
I may roll with my own solution, alternatively. It doesn't look like that one is full-featured enough (need to be able to retrieve views at indices)

**Comment by Tommy-LSA** [comment]
In NET we have very fast dictionaries (2 column collections). I mostly use this in parallel to save a key together with an index which points to the index of controlitem. While filling the control you have to keep the dictionary up2date. So you have very fast access to a position of control item without iterate the control items. The other direction could work similar.

**Comment by acomminos** [comment]
Just finished the draft design of my new treeview prototype. It's not very efficient right now, and doesn't maintain state as well, but it's a work in progress. Check it out upstream.

**Comment by acomminos** [comment]
Ok, much progress has been made towards the treeview. Any commits from c7093b96e35d87a10d6ada0dd0c1244c915e4cd9 should be ready for use as a daily driver. Feedback would be very much appreciated.

**Comment by acomminos** [comment]
Tree view is pretty much done at this point, with many optimizations. Code could still be cleaned up a bit.

Regardless, can anyone frequenting the aforementioned server please provide some feedback on how it works for you. Thanks!
